runner-from-prebacked-ami.tf

```diff
@@ -10,11 +10,12 @@ data "aws_ami" "custom_ami" {
   }
 }
 
-# create ec2 resourse 
+# create ec2 resourse
 resource "aws_instance" "gitlab-runner-prebacked" {
   count                       = "${var.use_prebacked_ami ? 1 : 0}"
   ami                         = "${data.aws_ami.custom_ami.id}"
   instance_type               = "${var.instance_type}"
+  key_name                    = "${aws_key_pair.key.key_name}"
```

Comment:
I think we should add a choice whether to spawn instance with or without ssh key.

Reply:
Ok, but then, ceration of aws_key_pair ressource should also depend on that choice? Currently it is always created.

Reply:
You can create SSH key with count depending on variable, same variable can be used here
e.g
`count  = "${var.create_ssh_key ? 1 :0)`

`key_name = "${var.create_ssh_key ? aws_key_pair.key.key_name : ""}"`

passing empty line as a key name should be checked, not sure if this will work.
If not  - I'm ok with ssh access to the instance by default

Reply:
@artyom-gatalsky  I've applied a bit different logic, check it please

Reply:
Hi @artyom-gatalsky have you had a chance to look into this?